Client/Frontend/Home/FirefoxHomeViewController.swift

```diff
@@ -493,15 +491,24 @@ extension FirefoxHomeViewController: UICollectionViewDelegateFlowLayout {
     }
 
     func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
-        let cellSize = Section(indexPath.section).cellSize(for: self.traitCollection, frameWidth: self.view.frame.width)
+        var cellSize = Section(indexPath.section).cellSize(for: self.traitCollection, frameWidth: self.view.frame.width)
 
         switch Section(indexPath.section) {
         case .topSites:
             // Create a temporary cell so we can calculate the height.
             let layout = topSiteCell.collectionView.collectionViewLayout as! HorizontalFlowLayout
             let estimatedLayout = layout.calculateLayout(for: CGSize(width: cellSize.width, height: 0))
             return CGSize(width: cellSize.width, height: estimatedLayout.size.height)
-        case .pocket, .recentlySaved:
+        case .recentlySaved:
+            if FxHomeRecentlySavedCollectionCell.recentlySavedItems.count > 8, UIDevice.current.userInterfaceIdiom == .pad {
+                cellSize.height *= 3
+                return cellSize
+            } else if FxHomeRecentlySavedCollectionCell.recentlySavedItems.count > 4, UIDevice.current.userInterfaceIdiom == .pad {
+                cellSize.height *= 2
+                return cellSize
+            }
```

Comment:
I know this is not ideal to calculate height like this as it will not have accessibility 
but I truly understand why you did it as I had to do the same

something to think in future if we can have dynamic height based on increased font size 🤔

Reply:
Hmm yeah or maybe use different cells (new design too for a11y maybe) when they get too large to support it in the current cell? I guess we can think about it more later.